src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -300,33 +305,87 @@ private static void SetProviderConfiguration(string strConfig, EventPipeConfigur
                 throw new ArgumentNullException(nameof(strConfig));
             }
 
-            // String must be of the form "providerName:keywords:level,providerName:keywords:level..."
-            string[] providers = strConfig.Split(ProviderConfigDelimiter);
+            // String must be of the form "providerName:keywords:level:parameter,providerName:keywords:level:parameter..."
+            // where parameter is of the form: "key=value;key=value..."
+            string[] providers = strConfig.Split(
+                ProviderConfigDelimiter,
+                StringSplitOptions.RemoveEmptyEntries); // Remove "empty" providers.
             foreach (string provider in providers)
             {
-                string[] components = provider.Split(ConfigComponentDelimiter);
-                if (components.Length == 3)
-                {
-                    string providerName = components[0];
+                const char FilterDataDelimiter = ';';
+                string[] components = provider.Split(
+                    ConfigComponentDelimiter,
+                    StringSplitOptions.None); // Keep empty tokens
```

Comment:
Should limit it to a maximum of 3 items returned.   That allows : in the Filter data.

Reply:
Based on feedback, I'm removing most error checking from here, so it's handle by callback.

---
In reply to: [230848452](https://github.com/dotnet/coreclr/pull/20796#discussion_r230848452) [](ancestors = 230848452)

Reply:
I think changing this to Split(ConfigComponentDelimiter, 3, StringSplitOptions.None) is a good change.  Otherwise if there is : in the parameters then anything after it will be ignored.